### DISCUSSION TITLE
How to set response directly in WebResourceRequested?

### DISCUSSION THREAD

**Discussion by roa-nyx** [MicrosoftEdge/WebView2Feedback#4024] [Q&A]
I'm trying to intercept certain requests in WebView2, and set their responses directly in the event handler. I've got the WebView2 successfully calling the callback for the URIs I'm interested in. Then I set up this event handler for WebResourceRequested:

```
		private void webView_WebResourceRequested(object? sender, CoreWebView2WebResourceRequestedEventArgs e)
		{
			if (e.Request.Method != "GET")
				throw new NotImplementedException();
			StreamWriter sw = new StreamWriter(ms, Encoding.UTF8, leaveOpen: true);
			sw.Write("HELLO WORLD!");
			sw.Flush();
			sw.Close();
			ms.Position = 0;
			e.Response = webView.CoreWebView2.Environment.CreateWebResourceResponse(ms, 200, "", "");
			e.Response.Headers.AppendHeader("Content-Type", "text/plain");
		}
```

`ms` is just a `new MemoryStream` declared at the class level. I know this is bad practice but I was concerned that maybe the stream was being disposed before CoreWebView actually streamed through the content, so I thought this might help with debugging for now.

I expected this to show "Hello World" in the webview however instead I get a plain white background with no text. Opening the dev tools and hitting Ctrl+R shows no request in the network tab.

This code does execute and the intellisense on `e.Response` includes this bit, which feels to me like this should work:

>		//     If this object is set, the Microsoft.Web.WebView2.Core.CoreWebView2.WebResourceRequested
>		//     event will be completed with this Response. An empty Microsoft.Web.WebView2.Core.CoreWebView2WebResourceResponse
>		//     object can be created with Microsoft.Web.WebView2.Core.CoreWebView2Environment.CreateWebResourceResponse(System.IO.Stream,System.Int32,System.String,System.String)
>		//     and then modified to construct the Response.

What am I missing here?

**Comment by novac42** [comment]
Thanks for reaching out. I've assigned this to a dev that can help follow up on this.

**Comment by roa-nyx** [comment]
Ok I figured it out.

Apparently the "ReasonPhrase" parameter must be "Ok" in this case in order to work. The corrected line below:

```
e.Response = webView.CoreWebView2.Environment.CreateWebResourceResponse(ms, 200, "Ok", "");
```

Not sure why edge doesn't register the request at all if this isn't set correctly. Can I suggest that there be some logging somewhere or an error thrown when there is a failure of this type?
